sources/packages/backend/apps/workers/src/services/application/application.service.ts

```diff
@@ -89,4 +109,148 @@ export class ApplicationService extends RecordDataModelService<Application> {
       },
     });
   }
+
+  /**
+   * Associates an MSFAA number to the application checking
+   * whatever is needed to create a new MSFAA or use an
+   * existing one instead.
+   * @param applicationId application id to receive an MSFAA.
+   */
+  async associateMSFAANumber(applicationId: number): Promise<Application> {
+    const application = await this.repo.findOne({
+      select: {
+        id: true,
+        applicationStatus: true,
+        student: {
+          id: true,
+        },
+        currentAssessment: {
+          id: true,
+          offering: {
+            id: true,
+            offeringIntensity: true,
+          },
+        },
+        msfaaNumber: {
+          id: true,
+        },
+      },
+      relations: {
+        student: true,
+        currentAssessment: {
+          offering: true,
+        },
+      },
+      where: {
+        id: applicationId,
+      },
+    });
+
+    if (!application) {
+      throw new CustomNamedError(
+        "Student Application not found or one of its associations is missing.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+
+    if (application.msfaaNumber?.id) {
```

Comment:
shouldn't `if (application.msfaaNumber) {` be enough

Reply:
Maybe, but I would prefer to use the id for this case since it indicates some DB value actually being tested.